### ISSUE TITLE
308 Redirects are not followed

### ISSUE THREAD

**Issue by ZanderBrown** [libgit2/libgit2#4675]
### Reproduction steps

Try to clone a GitLab repository like https://gitlab.com/esodan/gresg/ (including the final / and without .git) - I used gnome-builder to do this

### Expected behaviour

The repository is cloned after any necessary redirects

### Actual behaviour

Clone fails with error quoting status 308

### Version of libgit2 (release number or SHA1)

That used in `3.29.2-18-gb86e3ec flatpak-nightly` (not sure what that links to) but should affect all versions

### Operating system(s) tested

Fedora 28/flatpak

A 308 is defined as:
> **308 Permanent Redirect (RFC 7538)** The request and all future requests should be repeated using another URI. 307 and 308 parallel the behaviors of 302 and 301, but do not allow the HTTP method to change. So, for example, submitting a form to a permanently redirected resource may continue smoothly.

So basically it should be followed in the same way as 301, 302, 303 & 307

https://github.com/libgit2/libgit2/blob/408b16c1b06526f509fdb8997818d648d4d1ea1e/src/transports/http.c#L395-L398

[Originally reported @ GNOME Builder](https://gitlab.gnome.org/GNOME/gnome-builder/issues/521)

**Comment by carlosmn** [comment]
For reference [308 Permanent Redirect](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308) seems to be a newer code introduced to work around some libraries misusing [301 Moved Permanently](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301). Mozilla's documentation about the latter suggests using 308 for `POST` requests.

**Comment by ZanderBrown** [comment]
That's more about keeping the verb the same, 308 & GET is perfectly valid

**Comment by pks-t** [comment]
Hi @ZanderBrown, and sorry for not providing more feedback earlier. The fix for this issue should be trivially to do, as you've also already found the correct location where to implement it. Do you wish to give it a try yourself after all this time? Otherwise I'd take this over and implement it myself.

**Comment by ZanderBrown** [comment]
I've not built libgit2 before so it's probably faster if you do it

**Comment by ZanderBrown** [comment]
You win :-)

**Comment by carlosmn** [comment]
Fixed via #4848
